zmu_cortex_m/src/gdb/target.rs

```diff
@@ -0,0 +1,244 @@
+use gdbstub::target::Target;
+use gdbstub::target::TargetResult;
+use gdbstub::target;
+use gdbstub::target::ext::monitor_cmd::MonitorCmd;
+use gdbstub::common::Signal;
+use gdbstub::outputln;
+
+use crate::MemoryMapConfig;
+use crate::gdb::simulation;
+
+use gdbstub::target::ext::base::singlethread::SingleThreadBase;
+use gdbstub::target::ext::base::singlethread::SingleThreadResume;
+use gdbstub::target::ext::base::singlethread::SingleThreadSingleStep;
+use gdbstub::target::ext::base::singlethread::SingleThreadSingleStepOps;
+use gdbstub::target::ext::base::singlethread::SingleThreadResumeOps;
+use gdbstub::target::ext::base::singlethread::SingleThreadRangeStepping;
+use gdbstub::target::ext::base::singlethread::SingleThreadRangeSteppingOps;
+
+use crate::gdb::simulation::SimulationRunEvent;
+use crate::gdb::simulation::SimulationEvent;
+
+use crate::core::register::Reg;
+use crate::core::register::BaseReg;
+
+use crate::semihosting::SemihostingCommand;
+use crate::semihosting::SemihostingResponse;
+
+pub struct ZmuTarget {
+    simulation: simulation::Simulation,
+}
+
+impl ZmuTarget {
+    pub fn new(
+        code: &[u8],
+        semihost_func: Box<dyn FnMut(&SemihostingCommand) -> SemihostingResponse + 'static>,
+        map: Option<MemoryMapConfig>,
+        flash_size: usize,
+    ) -> ZmuTarget {
+        let simulation = simulation::Simulation::new(code, semihost_func, map, flash_size);
+        ZmuTarget {
+            simulation: simulation.unwrap(),
+        }
+    }
+ 
+    pub fn run(&mut self, poll_incomming_data: impl FnMut() -> bool ) -> SimulationRunEvent {
+        self.simulation.run(poll_incomming_data)
+    }
+
+    // pub fn reset(&mut self) -> Result<(), &'static str> {
```

Comment:
commented out function, maybe remove from this PR

Reply:
done